File: Outros/POO UML/Codigos/Aula 1/Exemplo cao/mundocao_simples/src/Cachorro.java
// Criando a classe cachorro
public class Cachorro {
    
    // Atributos
    public String raca;
    public int idade;
    public String corOlhos;
    public String corPelo;
    public String tipoPelo;

    // Comportamento
    public boolean Latir(String latido)
    {
        return true;
    }

    public void Correr()
    {
        System.out.println("Estou correndo 2 metros");
    }

    public void Dormir(int dormindoMinutos)
    {
        System.out.println("Estou dormindo por " + dormindoMinutos + " minuto(s)");
    }

    public void Comer()
    {
        System.out.println("Estou comendo...");
    }

    public void Acordar()
    {
        System.out.println("Estou acordando");
    }

}

File: Programação/Back-End PHP/04/mundocao/feitos em aula/abstracao/Cachorro.php
<?php

    class Cachorro {

        public string $nome;
        public string $raca;
        public int $idade;
        public string $corOlhos;
        public string $corPelo;

        // Comportamentos / Ações
        public function latir(string $latido){
            echo $latido . PHP_EOL;
        }

        public function comer() {
            echo "Estou comendo..!" . PHP_EOL;
        }

        public function dormir(){
            echo "Estou dormindo.." . PHP_EOL;
        }

        public function correr(){
            echo "Estou correndo.." . PHP_EOL;
        }

        public function acordar(){
            echo "Estou acordando.." . PHP_EOL;
        }

    }

?>